libcontainer/container_linux.go

```diff
@@ -1345,7 +1379,13 @@ func (c *linuxContainer) Restore(process *Process, criuOpts *CriuOpts) error {
 			req.Opts.InheritFd = append(req.Opts.InheritFd, inheritFd)
 		}
 	}
-	return c.criuSwrk(process, req, criuOpts, extraFiles)
+	err = c.criuSwrk(process, req, criuOpts, extraFiles)
+
+	for _, fd := range extraFiles {
```

Comment:
It's not clear whether this is coming from? Seems like a separate bugfix -- if so, can you make a separate commit so this fix will have an explanation in its commit message

Reply:
It's not really a separate bugfix. Before this change the FDs in extraFiles were opened in this function and immediately `defer Close`d. Now that the FDs are opened in another function `defer Close()` no longer works (to figure this out took me much longer than it should have :wink: ). So all the open FDs need to be closed after CRIU is done with it, that is why I am explicitly closing them all. Maybe a comment would make this clearer.

Reply:
Right.

*/me goes off to get another espresso*